design-documents/checkout/alternative-checkout-flow.md

```diff
@@ -0,0 +1,50 @@
+# Alternative checkout flow
+
+In scope of work on GraphQL and storefront APIs we have an opportunity to improve design and features of storefront checkout.
+
+The purpose of this document is to discuss possible alternatives to current Magento checkout flow. 
+
+  1. When Quote is created?
+     * For physical products on Review & Payments step
+     * For virtual products billing address has to be entered first
+  1. Cart properties:
+     * Line Items:
+       * SKU
+       * Selected options (custom/configurable)
+       * Quantity
+       * Regular price
+       * Price
+  4. Applicable cart rules calculator arguments
+     * Cart
+     * Addresses
```

Comment:
we need to pass some kind buyer representation here

Reply:
That would be the customer.